src/tasks/WasmAppBuilder/EmitWasmBundleFiles.cs

```diff
@@ -0,0 +1,270 @@
+        var verbose = remainingDestinationFilesToBundle.Length > 1;
+        var verboseCount = 0;
+
+        if (remainingDestinationFilesToBundle.Length > 0)
+        {
+            int allowedParallelism = Math.Max(Math.Min(remainingDestinationFilesToBundle.Length, Environment.ProcessorCount), 1);
+            if (BuildEngine is IBuildEngine9 be9)
+                allowedParallelism = be9.RequestCores(allowedParallelism);
+
+            Parallel.For(0, remainingDestinationFilesToBundle.Length, new ParallelOptions { MaxDegreeOfParallelism = allowedParallelism, CancellationToken = BuildTaskCancelled.Token }, i =>
+            {
+                var group = remainingDestinationFilesToBundle[i];
+
+                // Since the object filenames include a content hash, we can pick an arbitrary ITaskItem from each group,
+                // since we know each group's ITaskItems all contain the same binary data
+                var contentSourceFile = group.First();
+
+                var outputFile = group.Key;
+                var inputFile = contentSourceFile.ItemSpec;
+                if (verbose)
+                {
+                    var registeredName = contentSourceFile.GetMetadata("RegisteredName");
+                    if(string.IsNullOrEmpty(registeredName))
+                    {
+                        registeredName = Path.GetFileName(inputFile);
+                    }
+                    var count = Interlocked.Increment(ref verboseCount);
+                    Log.LogMessage(MessageImportance.Low, "{0}/{1} Bundling {2} ...", count, remainingDestinationFilesToBundle.Length, registeredName);
+                }
+
+                Log.LogMessage(MessageImportance.Low, "Bundling {0} as {1}", inputFile, outputFile);
+                var symbolName = ToSafeSymbolName(outputFile);
+                emitter(outputFile, (codeStream) => {
+                    using var inputStream = File.OpenRead(inputFile);
+                    BundleFileToCSource(symbolName, inputStream, codeStream);
+                });
+            });
+        }
+
+        var filesToBundleByRegisteredName = FilesToBundle.GroupBy(file => {
+            var registeredName = file.GetMetadata("RegisteredName");
+            if(string.IsNullOrEmpty(registeredName))
+            {
+                registeredName = Path.GetFileName(file.ItemSpec);
+            }
+            return registeredName;
+        }).ToList();
+
+        var files = filesToBundleByRegisteredName.Select(group => {
+            var registeredFile = group.First();
+            var outputFile = registeredFile.GetMetadata("DestinationFile");
+            var registeredName = group.Key;
+            var symbolName = ToSafeSymbolName(outputFile);
+            return (registeredName, symbolName);
+        }).ToList();
+
+        Log.LogMessage(MessageImportance.High, "Bundling {2} objects into mono_wasm_register_{0}_bundle as {1}", BundleName, BundleFile, files.Count);
```

Comment:
This message should be `Low` too. Instead at the beginning we can log a message like `Bundling {numFiles} files for {bundleName}`.